### PR TITLE
Add an inputfile for phase transition with Landau free energy

### LINKED ISSUE #9098: Add a test for structural phase transformation with Landau free energy
### Description of the enhancement or error report
Add a verification test for a structural phase transformation (multi-variants) with Landau type free energy.

### Rationale for the enhancement or information for reproducing the error
The free energy form is capable to describe the symmetry breaking during phase transformation.

### Identified impact
(i.e. Internal object changes, limited interface changes, public API change, or a list of specific applications impacted)
N/A.

### PR DESCRIPTION
**Pull request by yipenggao**
Add an input file for phase transition with Landau free energy (with multi-variants caused by symmetry breaking during phase transition). 

closes #9098

### Complete each of the following items before creating a PR

- Include any relevant design information for your change
- Follow [Coding Standards](http://mooseframework.org/wiki/CodeStandards/)
- Submit one or more [Test Cases](http://mooseframework.org/wiki/MooseTraining/testing/) or modify existing test cases
- Reference or close a specific issue (refs # or closes #)

### CODE REVIEW

**Comment by dschwen** [comment]
Please expand the file name. LD means very little.

**Comment by yipenggao** [comment]
@dschwen 

Yes. The file name is changed. Thanks.

File: modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+  zmax = 0
+  elem_type = QUAD4
+[]
+
+[Variables]
+  [./eta1]
+    order = FIRST
+    family = LAGRANGE
+    [./InitialCondition]
+      type = RndBoundingBoxIC
```
Comment (dschwen) [inline modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i]: any reason you are not just using a `RandomIC`? The box seems to cover the entire domain anyways.

File: modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+  ymax = 100
+  zmin = 0
+  zmax = 0
+  elem_type = QUAD4
+[]
+
+[Variables]
+  [./eta1]
+    order = FIRST
+    family = LAGRANGE
```
Comment (dschwen) [inline modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i]: FIRST/LAGRANGE are the default values

File: modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+  [../]
+  [./deta2dt]
+    type = TimeDerivative
+    variable = eta2
+  [../]
+[]
+
+#
+# Try visualizing the stress tensor components as done in Conserved.i
+#
```
Comment (dschwen) [inline modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i]: you can remove this comment

File: modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+    prop_values = '1 1'
+  [../]
+
+  [./chemical_free_energy]
+    type = DerivativeParsedMaterial
+    f_name = Fc
+    args = 'eta1 eta2'
+    constant_names = 'A2 A3 A4'
+    constant_expressions = '0.2 -12.6 12.4'
+    function = A2/2*(eta1^2+eta2^2)+A3/3*(eta1^3+eta2^3)+A4/4*(eta1^2+eta2^2)^2
```
Comment (dschwen) [inline modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i]: if you enclose the function in `''` you can use spaces to improve readability

File: modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+  [../]
+  [./matl_sigma22]
+    type = RankTwoAux
+    rank_two_tensor = stress
+    index_i = 0
+    index_j = 1
+    variable = sigma12_aux
+    execute_on = 'timestep_end initial'
+  [../]
+[]
```
Comment (dschwen) [inline modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i]: This is probably all copy pasted from the existing example (which I'm actually updating right now). There is a more elegant/compact way of doing this now.

File: modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+      [./right_x]
+    type = PresetBC
+    variable = disp_x
+    boundary = 'right'
+    value = 0
+  [../]
+  [./left_x]
+    type = PresetBC
+    variable = disp_x
+    boundary = 'left'
```
Comment (dschwen) [inline modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i]: you can specify multiple boundaries in a BC section, that will make this section musch smaller

File: modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i [hunk 7]
```diff
@@ -0,0 +1,301 @@
...
+  l_max_its = 30
+  nl_max_its = 10
+  l_tol = 1.0e-4
+  nl_rel_tol = 1.0e-8
+  nl_abs_tol = 1.0e-10
+  start_time = 0.0
+  num_steps = 10
+
+  [./TimeStepper]
+    type = SolutionTimeAdaptiveDT
```
Comment (dschwen) [inline modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i]: look at https://github.com/idaholab/moose/compare/devel...dschwen:examples_9011?expand=1 for my pending changes. We should be using `IterationAdaptiveDT` here.

File: modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i [hunk 8]
```diff
@@ -0,0 +1,301 @@
...
+    type = ComputeVariableEigenstrain
+    eigen_base = '-0.1 0.1 0 0 0 0'
+    prefactor = var_dep2
+    args = 'eta2'
+    eigenstrain_name = eigenstrain2
+  [../]
+
+  [./strain]
+    type = ComputeSmallStrain
+    displacements = 'disp_x disp_y'
```
Comment (dschwen) [inline modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i]: displacements should always be in GlobalParams (see link to my in-flight changes)

**Review by dschwen — CHANGES_REQUESTED** [review]

File: modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i (hunk 7 above)
Comment (yipenggao) [inline modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i]: @dschwen 
Thanks for your review. It has been updated according to your comments. (except the IterationAdaptiveDT, of which the convergence is much worse than as SolutionTimeAdaptiveDT, seems dt is greatly enlarged after each step, leading to non-convergence)

I will also revise other input file according to this.

Comment (dschwen) [inline modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i]: The growth and cutback factors can tune this. There is no reason IterationAdaptiveDT should perform intrinsically worse and it is more stable on systems with heterogeneous load.

**Comment by mangerij** [comment]
We also use the adaptive stepper for Landau problems with elasticity. Seems
to be most efficient

https://bitbucket.org/mesoscience/ferret/src/9f4532db4972a4630388998621b1df
ebb6abe788/inputs/PTOepitaxy_09_20_10_10_G1.i?at=master&
fileviewer=file-view-default

:)

File: modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i (hunk 7 above)
Comment (permcody) [inline modules/combined/examples/phase_field-mechanics/LandauPhaseTrans.i]: SolutionTimeAdaptive does not take predictable time step increments. It measures wall time to make decisions about step size so cannot be used for any kind of integration test.

**Comment by yipenggao** [comment]
Thanks all. It has been updated :).

**Review by dschwen — APPROVED** [review]
Looks good!

**Comment by dschwen** [comment]
I squashed the 4 commits into one.

**Pull request merged**